pkgs/development/libraries/catboost/default.nix

```diff
@@ -0,0 +1,135 @@
+{ lib
+, config
+, stdenv
+, fetchFromGitHub
+, autoPatchelfHook
+, cmake
+, libiconv
+, llvmPackages
+, ninja
+, openssl
+, python3Packages
+, ragel
+, yasm
+, zlib
+, cudaSupport ? config.cudaSupport or false
+, cudaPackages ? {}
+, pythonSupport ? false
+, symlinkJoin
+}:
+let
+  inherit (cudaPackages) cudaVersion;
+
+  cuda-common-redist = with cudaPackages; [
+    cuda_cccl
+    libcublas
+  ];
+  cuda-native-redist = symlinkJoin {
+    name = "cuda-native-redist-${cudaVersion}";
+    paths = with cudaPackages; [
+      cuda_cudart
+      cuda_nvcc
+    ] ++ cuda-common-redist;
+  };
+
+  cuda-redist = symlinkJoin {
+    name = "cuda-redist-${cudaVersion}";
+    paths = cuda-common-redist;
+  };
+in
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "catboost";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "catboost";
+    repo = "catboost";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-JcbGEkBQe4En/1g1BY8ytV7cWN9CjujMrThIrJ0o7dU=";
+  };
+
+  patches = [
+    ./remove-conan.patch
+  ];
+
+  postPatch = ''
+    substituteInPlace cmake/common.cmake \
+      --replace  "\''${RAGEL_BIN}" "${ragel}/bin/ragel" \
+      --replace "\''${YASM_BIN}" "${yasm}/bin/yasm"
+
+    shopt -s globstar
+    for cmakelists in **/CMakeLists.*; do
+      sed -i "s/OpenSSL::OpenSSL/OpenSSL::SSL/g" $cmakelists
+      ${lib.optionalString (lib.versionOlder cudaVersion "11.8") ''
+        sed -i 's/-gencode=arch=compute_89,code=sm_89//g' $cmakelists
+        sed -i 's/-gencode=arch=compute_90,code=sm_90//g' $cmakelists
+      ''}
+    done
+  '';
+
+  outputs = [ "out" "dev" ];
+
+  nativeBuildInputs = [
+    cmake
+    llvmPackages.bintools
+    ninja
+    (python3Packages.python.withPackages (ps: with ps; [ six ]))
+    ragel
+    yasm
+  ] ++ lib.optionals stdenv.isLinux [
+    autoPatchelfHook
+  ] ++ lib.optionals cudaSupport [
+    cuda-native-redist
+  ];
+
+  buildInputs = [
+    openssl
+    zlib
+  ] ++ lib.optionals stdenv.isDarwin [
+    libiconv
+  ] ++ lib.optionals cudaSupport [
+    cuda-redist
```

Comment:
Is symlinkJoin necessary?

Reply:
It may not be necessary.
I found this approach in other packages, so I just followed the pattern.

Reply:
> I found this approach in other packages, so I just followed the pattern.

The `symlinkJoin` trick is still necessary for packages that use `FindCUDA.cmake`, but the preferred way is to use `FindCUDAToolkit.cmake` and avoid `symlinkJoin`. One of the costs of `symlinkJoin` is increased runtime closure size

Reply:
Thanks, that makes sense.